py/kubeflow/testing/argo_client.py

```diff
@@ -16,10 +16,16 @@
 
 def log_status(workflow):
   """A callback to use with wait_for_workflow."""
-  logging.info("Workflow %s in namespace %s; phase=%s",
-           workflow["metadata"]["name"],
-           workflow["metadata"]["namespace"],
-           workflow["status"]["phase"])
+  try:
+    logging.info("Workflow %s in namespace %s; phase=%s",
+                 workflow["metadata"]["name"],
+                 workflow["metadata"]["namespace"],
+                 workflow["status"]["phase"])
+  except KeyError as e:
+    # Ignore the error and just log the stacktrace
+    # as sometimes the workflow object does not have all the fields
```

Comment:
do we know why it "sometimes" doesn't have that field?
sounds pretty weird

Reply:
Should we log an issue and reference here?

Reply:
Linked the tracking issue in the comment. Probably don't need to log it.